File: Web Application/Backend/repository/question.go
package repository

import (
	"context"
	"database/sql"
	"time"
)

type Question struct {
	ID            string
	Question      string
	DurationLimit int
	Deleted       bool
	CreatedAt     time.Time
	UpdatedAt     sql.NullTime
	DeletedAt     sql.NullTime
	Labels        []*QuestionLabel
}

type QuestionLabel struct {
	ID 						string
	QuestionID 		string
	CompetencyID 	string
	Deleted 			bool
	CreatedAt 		time.Time
	UpdatedAt 		sql.NullTime
	DeletedAt 		sql.NullTime
}

type Labels struct {
	IDs 					[]string
	CompetencyIDs []string
}

type QuestionRepository interface {
	Insert(context.Context, *Question, *Labels) error
	SelectAll(context.Context) ([]*Question, error)
	SelectAllByRoomID(context.Context, string) ([]*Question, error)
	SelectOneByID(context.Context, string) (*Question, error)
	Upsert(context.Context, *Question, *Labels) error
	DeleteByID(context.Context, string) error
}

File: Web Application/Frontend/src/interface/question.ts
export interface QuestionLabel {
  id: string,
  competency_id: string,
}

export interface Question {
  id: string,
  question: string,
  duration_limit: number,
  org_position: string,
  start_answer: string,
  transcript?: string
  labels: QuestionLabel[]
}

export interface QuestionLabelOptions {
  id : string,
  competency: string
}